amass/shodan.go

```diff
@@ -16,7 +16,7 @@ type Shodan struct {
 	BaseService
 
 	SourceType string
-	APIKey     *APIKey
+	API        *APIKey
```

Comment:
I think we should standardize how API credentials should be stored/scoped. Looks like most of the sources  store the credentials in a function local var while this one stores it as a member of the service struct.